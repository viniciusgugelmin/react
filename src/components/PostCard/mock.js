export const PostCardPropsMock = {
  title: 'Lorem ipsum title',
  body: 'Lorem ipsum body',
  id: 1,
  cover: 'img/lorem-ipsum-img.png',
};
