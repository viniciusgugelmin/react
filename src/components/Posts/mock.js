export const PostsMock = [
  {
    id: 1,
    title: 'Lorem ipsum title 1',
    body: 'Lorem ipsum body 1',
    cover: 'img/img1.png',
  },
  {
    id: 2,
    title: 'Lorem ipsum title 2',
    body: 'Lorem ipsum body 2',
    cover: 'img/img2.png',
  },
  {
    id: 3,
    title: 'Lorem ipsum title 3',
    body: 'Lorem ipsum body 3',
    cover: 'img/img3.png',
  },
];
