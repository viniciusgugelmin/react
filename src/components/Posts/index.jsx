import './styles.css';

import { PostCard } from '../PostCard';

export const Posts = ({ posts = [] }) => (
  <div className="posts">
    {posts.length > 0
      ? posts.map(post => <PostCard key={post.id} cover={post.cover} title={post.title} body={post.body} />)
      : 'No posts found'}
  </div>
);
