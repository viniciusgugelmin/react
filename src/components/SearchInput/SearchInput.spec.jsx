import { SearchInput } from './index';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Posts } from '../Posts';
import { PostsMock as posts } from '../Posts/mock';

describe('<SearchInput />', () => {
  it('should have a value of searchValue', () => {
    expect.assertions(2);

    const fn = jest.fn();
    const searchValue = 'Lorem ipsum';
    render(<SearchInput handleChange={fn} searchValue={searchValue} />);

    const input = screen.getByPlaceholderText(/^Type to search...$/i);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe(searchValue);
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/^Type to search...$/i);
    const value = 'Lorem ipsum';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapchot', () => {
    expect.assertions(1);

    const fn = jest.fn();
    const { container } = render(<SearchInput handleChange={fn} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
