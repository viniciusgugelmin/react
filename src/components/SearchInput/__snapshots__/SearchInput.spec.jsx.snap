// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapchot 1`] = `
<label>
  <input
    class="search-input"
    placeholder="Type to search..."
    type="search"
    value=""
  />
</label>
`;
